include/deal.II/grid/grid_tools.h

```diff
@@ -1438,7 +1515,31 @@ namespace GridTools
         }
   }
 
-
+  template <class Container, typename Predicate>
+  void
+  compute_active_cell_halo_layer (const Container                                       &container,
+                                  const Predicate                                       &predicate,
+                                  std::vector<typename Container::active_cell_iterator> &active_halo_layer)
+  {
+    active_halo_layer.clear ();
+    std::vector<bool> locally_active_vertices_on_subdomain (get_tria(container).n_vertices(),
+                                                            false);
+
+    for (typename Container::active_cell_iterator
+         cell = container.begin_active();
+         cell != container.end(); ++cell)
+      if (predicate(cell)) // True predicate --> Subdomain
+        for (unsigned int v=0; v<GeometryInfo<Container::dimension>::vertices_per_cell; ++v)
+          locally_active_vertices_on_subdomain[cell->vertex_index(v)] = true;
+
+    for (typename Container::active_cell_iterator
+         cell = container.begin_active();
+         cell != container.end(); ++cell)
+      if (!predicate(cell)) // False predicate --> Potential halo cell
+        for (unsigned int v=0; v<GeometryInfo<Container::dimension>::vertices_per_cell; ++v)
+          if (locally_active_vertices_on_subdomain[cell->vertex_index(v)] == true)
+            active_halo_layer.push_back(cell);
```

Comment:
You'll need a `break` here or you may push the same cell to the end multiple times.